### ISSUE TITLE
rpmdeps inspection does not handle macros in explicit dep names

### ISSUE THREAD

**Issue by dcantrell** [rpminspect/rpminspect#688]
Compare:

elfutils-0.186-1.el9
elfutils-0.186-3.el9

Any arch and you can see.  Since debuginfo packages are automatic, rpmdeps should probably just ignore debuginfo packages.

**Comment by jimbair** [comment]
Per the reporter in IRC (to keep track of the context):

```
 Not having a github account I am unable to comment there. But I do want to note that the package is called elfutils-debuginfod which is the debuginfod server, not the debuginfo elfutils package.
 Maybe that was obvious. But I have seen some confusion about the name (and in the past errata tool had a bug which assumed packages with debuginfo in their name were all debuginfo sub-packages).
```

**Comment by dcantrell** [comment]
This issue has nothing to do with debuginfo packages.  This build has a subpackage with 'debuginfod' in the name, which is the debuginfo daemon.

The finding by the rpmdeps inspection is showing up because the inspection does not handle macros in the explicit dep names.  This is a result seen:

```
1) Subpackage elfutils-debuginfod on aarch64 carries 'Requires: libdebuginfod.so.1()(64bit)' which comes from subpackage elfutils-debuginfod-client but does not carry an explicit package version requirement.  Please add 'Requires: elfutils-debuginfod-client = %{version}-%{release}' to the spec file to avoid the need to test interoperability between various combinations of old and new subpackages.

Result: VERIFY
Waiver Authorization: Anyone

Suggested Remedy:
Add the indicated explicit Requires to the spec file for the named subpackage.  Subpackages depending on shared libraries in another subpackage must carry an explicit 'Requires: SUBPACKAGE_NAME = %{version}-%{release}' in the spec file.
```

But the spec file has this for the elfutils-debuginfod subpackage:

```
Requires: elfutils-debuginfod-client%{depsuffix} = %{version}-%{release}
```

rpminspect doesn't know what to do with the %{depsuffix} macro, so it fails.

**Comment by dcantrell** [comment]
I've got the last of this fixed up in PR #702
